src/overlays/actors/ovl_Object_Kankyo/z_object_kankyo.c

```diff
@@ -33,42 +38,689 @@ const ActorInit Object_Kankyo_InitVars = {
+            break;
+
+        case 4:
+            func_808DC038(this, globalCtx);
+            break;
+    }
+}
+
+void ObjectKankyo_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    ObjectKankyo* this = THIS;
+
+    Actor_MarkForDeath(&this->actor);
+}
+
+void func_808DC18C(ObjectKankyo* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s32 pad2;
+    s32 pad3;
+    f32 sqrt;
+    f32 y;
+    f32 z;
+    f32 temp_f18;
+    Vec3f sp30;
+    f32 x;
+    f32 sp1C;
+
+    x = globalCtx->view.at.x - globalCtx->view.eye.x;
+    y = globalCtx->view.at.y - globalCtx->view.eye.y;
+    z = globalCtx->view.at.z - globalCtx->view.eye.z;
+    sqrt = sqrtf(SQ(x) + SQ(y) + SQ(z));
+
+    temp_f18 = x / sqrt;
+    x = z / sqrt;
+    sp1C = (y / sqrt) * 120.0f;
+
+    this->unk_14C[0].unk_00 = globalCtx->view.eye.x + (temp_f18 * 50.0f);
+    this->unk_14C[0].unk_04 = globalCtx->view.eye.y + sp1C;
+    this->unk_14C[0].unk_08 = globalCtx->view.eye.z + (x * 50.0f);
+    this->unk_14C[1].unk_00 = globalCtx->view.eye.x + (temp_f18 * 220.0f);
+    this->unk_14C[1].unk_08 = globalCtx->view.eye.z + (x * 220.0f);
+    this->unk_114C = 0;
+    this->unk_144 = 100.0f;
+
+    if ((this->unk_14C[0].unk_00 < -252.0f) && (this->unk_14C[0].unk_00 > -500.0f)) {
+        if ((this->unk_14C[0].unk_08 > 3820.0f) && (this->unk_14C[0].unk_08 < 4150.0f)) {
+            this->unk_114C = 1;
+            this->unk_144 = 400.0f;
+            if (x < 0.0f) {
+                this->unk_14C[0].unk_00 = -350.0f;
+                this->unk_14C[0].unk_04 = globalCtx->view.eye.y + sp1C;
+                this->unk_14C[0].unk_08 = 3680.0f;
+                this->unk_14C[1].unk_00 = -350.0f;
+                this->unk_14C[1].unk_08 = 3680.0f;
+            } else {
+                this->unk_14C[0].unk_00 = -350.0f;
+                this->unk_14C[0].unk_04 = globalCtx->view.eye.y + sp1C;
+                this->unk_14C[0].unk_08 = 4280.0f;
+                this->unk_14C[1].unk_00 = -350.0f;
+                this->unk_14C[1].unk_08 = 4280.0f;
+            }
+        }
+    }
+
+    sqrt = globalCtx->envCtx.windSpeed / 60.0f;
+    sqrt = CLAMP(sqrt, 0.0f, 1.0f);
+
+    sp30.x = globalCtx->envCtx.windDir.x * sqrt;
+    sp30.y = globalCtx->envCtx.windDir.y + 100.0f;
+    sp30.z = globalCtx->envCtx.windDir.z * sqrt;
+    this->unk_14C[2].unk_00 = 0x4000 - Math_Vec3f_Pitch(&D_801D15B0, &sp30);
+    this->unk_14C[2].unk_04 = Math_Vec3f_Yaw(&D_801D15B0, &sp30) + 0x8000;
+}
+
+void func_808DC454(ObjectKankyo* this, GlobalContext* globalCtx) {
+    s16 i;
+    u32 tempI;
+    f32 phi_f20;
+    f32 spD0;
+    f32 spCC;
+    f32 spC8;
+    f32 spC4;
+    f32 spC0;
+    f32 spBC;
+    f32 temp_f0_4;
+    f32 temp_f22;
+    f32 temp_f24;
+    f32 temp_f28;
+    f32 x = globalCtx->view.at.x - globalCtx->view.eye.x;
+    f32 y = globalCtx->view.at.y - globalCtx->view.eye.y;
+    f32 z = globalCtx->view.at.z - globalCtx->view.eye.z;
+    f32 sqrt = sqrtf(SQ(x) + SQ(y) + SQ(z));
```

Comment:
Same here, I agree that `sqrt` is too close to built-in functions and doesn't make for a good temp. `magnitude` is fine here